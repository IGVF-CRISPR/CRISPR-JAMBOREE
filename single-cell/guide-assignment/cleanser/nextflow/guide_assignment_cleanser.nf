process guide_assignment_cleanser {
    input:
        path mudata_input
        path mudata_output
        val method
        val threshold
    output:
        path mudata_output, emit: mudata_output

    script:
        def thresh_opt = threshold ? "-t ${threshold}" : ""
        """
            python ${moduleDir}/bin/igvf_guide_assignment.py  -i ${mudata_input} -o ${mudata_output} ${thresh_opt} --${method}
        """
}
