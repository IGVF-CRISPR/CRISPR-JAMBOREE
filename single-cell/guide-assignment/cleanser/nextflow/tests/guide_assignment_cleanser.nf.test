nextflow_process {
    name "Test Process guide_assignment_cleanser"
    script "guide_assignment_cleanser.nf"
    process "guide_assignment_cleanser"

    test("Cleanser guide assignment on Gasperini data") {
        when {
            process {
                """
                input[0] = file("${projectDir}/data/gasperini_guide_assignment_input_minimal.h5mu")
                input[1] = file("mudata_out.h5mu")
                input[2] = "cleanser"
                input[3] = false
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
                assert mudata_output != null
            }
        }
    }

    test("Threshold guide assignment on Gasperini data") {
        when {
            process {
                """
                input[0] = file("${projectDir}/data/gasperini_guide_assignment_input_minimal.h5mu")
                input[1] = file("mudata_out.h5mu")
                input[2] = "umi-threshold"
                input[3] = 5
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
                assert mudata_output != null
            }
        }
    }

    test("Cleanser guide assignment on Papalexi data") {
        when {
            process {
                """
                input[0] = file("${projectDir}/data/papalexi_guide_assignment_input.h5mu")
                input[1] = file("mudata_out.h5mu")
                input[2] = "cleanser"
                input[3] = false
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
                assert mudata_output != null
            }
        }
    }

    test("Threshold guide assignment on Papalexi data") {
        when {
            process {
                """
                input[0] = file("${projectDir}/data/papalexi_guide_assignment_input.h5mu")
                input[1] = file("mudata_out.h5mu")
                input[2] = "umi-threshold"
                input[3] = 5
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
                assert mudata_output != null
            }
        }
    }
}
