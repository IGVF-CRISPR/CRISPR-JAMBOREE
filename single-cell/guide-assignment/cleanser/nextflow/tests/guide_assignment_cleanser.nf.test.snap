{
    "guide_assignment_cleanser on Gasperini data": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "out": {
                    "0": [
                        "mudata_out.h5mu:md5,aa1fb5f4074250c238b86000f4d3c501"
                    ]
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 1,
                    "tasksSucceeded": 1
                },
                "name": "guide_assignment_cleanser",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-17T10:55:04.410916"
    },
    "guide_assignment_cleanser on Papalexi data": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "out": {
                    "0": [
                        "mudata_out.h5mu:md5,0b8fc6649f7ae28713787326591e726a"
                    ]
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 1,
                    "tasksSucceeded": 1
                },
                "name": "guide_assignment_cleanser",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-17T10:55:13.567054"
    }
}